src/foam/nanos/http/AuthWebAgent.java

```diff
@@ -0,0 +1,106 @@
+/**
+ * @license
+ * Copyright 2017 The FOAM Authors. All Rights Reserved.
+ * http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package foam.nanos.http;
+
+import foam.core.*;
+import foam.nanos.http.ProxyWebAgent;
+import foam.nanos.http.WebAgent;
+import java.io.PrintWriter;
+import foam.nanos.auth.AuthService;
+import javax.servlet.http.Cookie;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.util.Date;
+import foam.nanos.session.Session;
+import foam.dao.DAO;
+import foam.nanos.auth.User;
+
+public class AuthWebAgent
+  extends ProxyWebAgent
+{
+  public AuthWebAgent(WebAgent delegate) {
+    setDelegate(delegate);
+  }
+
+  public void execute(X x) {
+    HttpServletRequest  req  = (HttpServletRequest) x.get(HttpServletRequest.class);
+    HttpServletResponse resp = (HttpServletResponse) x.get(HttpServletResponse.class);
+    PrintWriter         out  = (PrintWriter) x.get(PrintWriter.class);
+
+    DAO                 sessionDAO = (DAO) x.get("sessionDAO");
+
+    Cookie[] cookies = req.getCookies();
+    String sessionId = null;
+    if( cookies != null ){
+      for ( Cookie cookie : cookies ) {
+        if ( cookie.getName().toString().equals("sessionId" )){
+          sessionId = cookie.getValue().toString();
+          Session session = (Session) sessionDAO.find(sessionId);
+          Date expiry = session.getExpiry();
+          Date dateNow = new Date();
```

Comment:
You don't need to worry about expiry because there's a cron job to remove expired sessions.